aim2dat/strct/strct_coordination.py

```diff
@@ -104,44 +78,103 @@ def calc_stdev(values):
             dist_matrix,
         ] + method_args[method]
 
-    for el_idx in range(len(structure["elements"])):
-        coord_numbers.append(method_function(el_idx, *method_args))
-    numbers = {}
-    distances = {}
-    weights = {}
-    for el1, coord0 in zip(structure["elements"], coord_numbers):
+    method_function = globals()["_coord_calculate_" + method]
+    sites = []
+    for site_idx in range(len(structure["elements"])):
+        sites.append(method_function(site_idx, *method_args))
+    stat_keys = ["distance"]
+    is_optional = [False]
+    if method == "voronoi" and voronoi_weight_type is not None or method == "econ":
+        stat_keys.append("weight")
+        is_optional.append(False)
+    coordination = _calculate_statistical_quantities(structure, sites, stat_keys, is_optional)
+    coordination["sites"] = sites
+    return None, coordination
+
+
+def _coord_group_method_args(
+    min_dist_delta,
+    n_nearest_neighbours,
+    econ_tolerance,
+    econ_conv_threshold,
+    voronoi_weight_type,
+    voronoi_weight_threshold,
+):
+    return {
+        "minimum_distance": [min_dist_delta],
+        "n_nearest_neighbours": [n_nearest_neighbours],
+        "econ": [econ_tolerance, econ_conv_threshold],
+        "voronoi": [voronoi_weight_type, voronoi_weight_threshold],
+    }
+
+
+def _create_site_dict(structure, site_idx, neighbours, weights=None):
+    """
+    Create dictionary storing information for a specific site.
+    """
+    site_dict = {el0: 0 for el0 in structure._element_dict.keys()}
+    site_dict["element"] = structure.elements[site_idx]
+    site_dict["kind"] = None if structure.kinds is None else structure.kinds[site_idx]
+    site_dict["position"] = list(structure.positions[site_idx])
+    site_dict["neighbours"] = []
+
+    distances = []
+    for idx, (neigh_idx, neigh_pos, distance) in enumerate(neighbours):
+        neigh_dict = {
+            "element": structure.elements[neigh_idx],
+            "kind": None if structure.kinds is None else structure.kinds[neigh_idx],
+            "site_index": neigh_idx,
+            "distance": float(distance),
+            "position": [float(pos) for pos in neigh_pos],
+        }
+        if weights is not None and len(weights) > 0:
```

Comment:
If I'm not missing something, this could be simplified to:

```suggestion
        if weights:
```

In case I'm missing something, leave it as it is